Shell/main.cpp

```diff
@@ -300,16 +300,41 @@ static Vector<String> expand_globs(const StringView& path, const StringView& bas
     return res;
 }
 
+static Vector<String> expand_parameters(const StringView& param)
```

Comment:
I don't understand why this function returns a `Vector` and not just a `String`?

Reply:
It returns a vector so that the environment variables result can be expanded by the `expand_globs` function, rather than just skipping over expanding globs for environment variables it allows them to be parsed. This is to mimic the behavior of bash, when you do:
```
$ test_var=*
$ echo $test_var
Desktop Documents Downloads
$ cd Desktop
$ echo $test_var
Notes Programming
```
Rather than being set to the current value, it changes when you cd away.
`expand_globs` would work for this env-variable, if you just passed in the string to it, but it wouldn't work for more complicated ones. Such as: `test_var="* *"` Which would duplicate the output, but if it's not separate than expand_globs wouldn't.